android/java/res/layout/radio_button_group_homepage_preference.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
+<!-- Copyright (c) 2025 The Brave Authors. All rights reserved.
+     This Source Code Form is subject to the terms of the Mozilla Public
+     License, v. 2.0. If a copy of the MPL was not distributed with this file,
+     You can obtain one at https://mozilla.org/MPL/2.0/. -->
+
+<LinearLayout
+    xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:orientation="vertical">
+
+  <!-- TODO(crbug.com/40672235): Change to the exact style TextMessagePreference is using. -->
+  <TextView
+      android:id="@+id/title"
+      style="@style/PreferenceTitle"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:paddingStart="@dimen/homepage_preference_title_lateral_padding"
+      android:paddingTop="@dimen/homepage_preference_title_vertical_padding"
+      android:paddingEnd="@dimen/homepage_preference_title_lateral_padding"
+      android:paddingBottom="@dimen/homepage_preference_title_vertical_padding"
+      android:text="@string/options_homepage_edit_label" />
+
+  <org.chromium.components.browser_ui.widget.RadioButtonWithDescriptionLayout
+      android:id="@+id/radio_button_group"
+      android:layout_width="match_parent"
+      android:layout_height="wrap_content"
+      android:focusableInTouchMode="true">
+
+    <!-- TODO(crbug.com/40117411): Remove android:background for radio buttons here and below after enhancement. -->
+    <org.chromium.components.browser_ui.widget.RadioButtonWithDescription
+        android:id="@+id/radio_button_chrome_ntp"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:primaryText="@string/options_homepage_chrome_homepage" />
+
+    <org.chromium.components.browser_ui.widget.RadioButtonWithDescription
+        android:id="@+id/brave_homepage_mobile_bookmarks"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        app:primaryText="@string/brave_homepage_mobile_bookmarks_text" />
+
+    <org.chromium.components.browser_ui.widget.RadioButtonWithEditText
+        android:id="@+id/radio_button_uri_edit"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content"
+        android:hint="@string/options_homepage_edit_hint"
+        android:inputType="textUri" />
+
+  </org.chromium.components.browser_ui.widget.RadioButtonWithDescriptionLayout>
+</LinearLayout>
```

Comment:
I would suggest adding `fix,-indent-with-non-tab,trailing-space,cr-at-eol` to `core.whitespace` in your gitconfig

Reply:
Addressed here fa4e1e7cec354525750a0db9aea4d922884dca56